waspc/data/Generator/templates/sdk/wasp/client/operations/rpc.ts

```diff
@@ -0,0 +1,38 @@
+import { type Route } from "wasp/client";
+import type {
+  _Awaited,
+  _ReturnType,
+} from "wasp/universal/types"
+
+// PRIVATE API (but should maybe be public, users define values of this type)
+export type Query<Input, Output> = QueryFunction<Input, Output> & QueryMetadata
+
+// PRIVATE API (but should maybe be public, users define values of this type)
+export type Action<Input, Output> = ClientOperation<Input, Output>
```

Comment:
@infomiho What do you think? Should these types be added to our public API?

Won't do this now, but I can create an issue.

Reply:
Done: https://github.com/wasp-lang/wasp/issues/2020